cpp/ql/test/examples/BadLocking/UnintendedDeclaration.cpp

```diff
@@ -0,0 +1,93 @@
+
+class Mutex
+{
+public:
+	Mutex();
+	~Mutex();
+
+	void lock();
+	void unlock();
+
+private:
+	// ...
+};
+
+template<class T>
+class Lock
+{
+public:
+	Lock() : m(0)
+	{
+	}
+
+	Lock(T &_m) : m(&_m)
+	{
+		m->lock();
+	}
+
+	~Lock()
+	{
+		m->unlock();
```

Comment:
If you guard this in `if (m)`, then none of the bad examples below will segfault. That makes it a much more scary bug.

Reply:
Updated.